### PR TITLE
Feature/inject product summary

### PR DESCRIPTION
**Pull request by ClaudivanFilho**
#### What is the purpose of this pull request?
Now the Shelf Component can import an ExtensionPoint instead a ShelfItem Component

#### What problem is this solving?
- Inject a ProductSummary extension inside of shelf
- Custom CSS classes

#### How should this be manually tested?
https://shelf-fail--storecomponents.myvtex.com/

#### Screenshots or example usage
screenshot from 2018-04-18 11-45-25

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# editorconfig.org
```
Comment (andreldsa) [inline .editorconfig]: Isn't to be ignored?

File: README.md [hunk 2]
```diff
@@ -41,9 +41,21 @@ Add "vtex.shelf" as your app dependency.
...
+## CSS Classes
+
+```css
+vtex-shelf
+vtex-shelf__title-text
+vtex-shelf__title-content
+vtex-shelf__slide
+vtex-shelf__arrow-right
+vtex-shelf__arrow-left
+vtex-shelf__dots
```
Comment (brunojdo) [inline README.md]: `vtex-shelf__arrow-right--disabled`
`vtex-shelf__arrow-left--disabled`

File: react/Shelf.js [hunk 3]
```diff
@@ -132,7 +185,7 @@ Shelf.schema = {
 Shelf.defaultProps = {
   maxItems: 10,
   scroll: 'BY_PAGE',
-  arrows: true
+  arrows: true,
```
Comment (brunojdo) [inline react/Shelf.js]: this comma is right?

File: react/ShelfItem.js [hunk 4]
```diff
@@ -6,28 +6,29 @@ import PropTypes from 'prop-types'
...
+    const { name, imageUrl, listPrice } = this.props
+
     return (
-      <div className="items-center flex flex-column pointer">
-        <h4 className="w-90 tc truncate mid-gray" title={productName}>
-          {productName}
-        </h4>
-        <img height={imageHeight} src={imagePath.replace('http:', '')} />
-        <h5 className="near-black">R$ {price.toFixed(2)}</h5>
+      <div className="ph4 grow ">
```
Comment (brunojdo) [inline react/ShelfItem.js]: Isn't here that product-summary will be injected?

File: .editorconfig (hunk 1 above)
Comment (ClaudivanFilho) [reply .editorconfig]: I don't think so, look this repository -> https://github.com/vtex-apps/dreamstore

File: README.md (hunk 2 above)
Comment (ClaudivanFilho) [reply README.md]: @brunojdo the react-slick sets dynamically a class named 'slick-disabled' to when the arrow is disabled. If i want to modify the disable state of an arrow, the only way is to override the selector ```.vtex-shelf__arrow-right .slick-disabled```

File: react/Shelf.js (hunk 3 above)
Comment (ClaudivanFilho) [reply react/Shelf.js]: @brunojdo the eslint warns about this comma

File: react/ShelfItem.js (hunk 4 above)
Comment (ClaudivanFilho) [reply react/ShelfItem.js]: No, this component is never used, i need to remove it :x

Comment (brunojdo) [reply react/ShelfItem.js]: So, remove it. 😄

**Pull request merged**